mininet/link.py

```diff
@@ -100,8 +115,18 @@ def updateIP( self ):
         ifconfig, _err, _exitCode = self.node.pexec(
             'ifconfig %s' % self.name )
         ips = self._ipMatchRegex.findall( ifconfig )
+        ip6s = self._ip6MatchRegex.findall( ifconfig )
         self.ip = ips[ 0 ] if ips else None
-        return self.ip
+        self.ip6 = None
+        self.ip6ll = None
+        for addr, scope in ip6s:
+            if scope == 'link':
+                if self.ip6ll is None:
+                    self.ip6ll = addr
+            else:
+                if self.ip6 is None:
+                    self.ip6 = addr
+        return self.IP()
```

Comment:
Instead of return `self.ip`, return `self.IP()`, the updated version of which is a little smarter.